.github/workflows/ibis-flink-backend.yml

```diff
@@ -0,0 +1,126 @@
+name: Flink Backend
+
+on:
+  push:
+    # Skip the backend suite if all changes are docs
+    paths-ignore:
+      - "docs/**"
+      - "**/*.md"
+      - "**/*.qmd"
+      - "codecov.yml"
+      - ".envrc"
+    branches:
+      - master
+      - "*.x.x"
+  pull_request:
+    # Skip the backend suite if all changes are docs
+    paths-ignore:
+      - "docs/**"
+      - "**/*.md"
+      - "**/*.qmd"
+      - "codecov.yml"
+      - ".envrc"
+    branches:
+      - master
+      - "*.x.x"
+  merge_group:
+
+permissions:
+  # this allows extractions/setup-just to list releases for `just` at a higher
+  # rate limit while restricting GITHUB_TOKEN permissions elsewhere
+  contents: read
+
+concurrency:
+  group: ${{ github.repository }}-${{ github.head_ref || github.sha }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+env:
+  FORCE_COLOR: "1"
+
+jobs:
+  test_flink_backend:
+    name: Flink ${{ matrix.os }} python-${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+        python-version:
+          - "3.9"
+        backend:
+          - name: flink
+            extras:
+              - flink
+            additional_deps:
+              - apache-flink
+              - grpcio-status # FIXME(deepyaman)
+            even_more_deps:
+              - pandas~=1.5
```

Comment:
Note that all this can be done in a cleaner way, if not trying to fit into the rest of the backends. Again, clean up can be done once things "work".